syntax = "proto3";
package im;

message MessageContent {
    int32  type         = 1;
    string data         = 2; // 文本可放
    string mime_type    = 3;
    string fid          = 4;
}

message MessageItem {
    string id        = 1;
    int32  to_id     = 2;
    int32  from_id   = 3;
    string  timestamp = 4; // unix ms
    int32  env       = 5;
    MessageContent content =6;
}
